b2share/modules/records/views.py

```diff
@@ -305,21 +357,77 @@ def put(*args, **kwargs):
         """Disable PUT."""
         abort(405)
 
-    def delete(self, *args, **kwargs):
+    @pass_record
+    def delete(self, pid, record, *args, **kwargs):
         """Delete a record."""
+        self.check_etag(str(record.model.version_id))
         pid_value = request.view_args['pid_value']
         pid, record = pid_value.data
 
         # Check permissions.
         permission_factory = self.delete_permission_factory
         if permission_factory:
             verify_record_permission(permission_factory, record)
+        record.delete()
+        db.session.commit()
+        return '', 204
 
-        delete_record(pid, record)
 
-        db.session.commit()
+class RecordsVersionsResource(ContentNegotiatedMethodView):
+
+    view_name = '{0}_versions'
 
-        return 'Record was deleted.', 200
+    def __init__(self, resolver=None, **kwargs):
+        """Constructor.
+
+        :param resolver: Persistent identifier resolver instance.
+        """
+        default_media_type = 'application/json'
+        super(RecordsVersionsResource, self).__init__(
+            serializers={
+                'application/json': lambda response: jsonify(response)
+            },
+            default_method_media_type={
+                'GET': default_media_type,
+            },
+            default_media_type=default_media_type,
+            **kwargs)
+        self.resolver = resolver
+
+    def get(self, pid=None, **kwargs):
+        """GET record versions."""
+        record_endpoint = 'b2share_records_rest.{0}_item'.format(
+            RecordUUIDProvider.pid_type)
+
+        pid_value = request.view_args['pid_value']
+        pid = RecordUUIDProvider.get(pid_value).pid
+        pid_versioning = PIDVersioning(child=pid)
+        if pid_versioning.is_child:
+            version_parent_pid_value = pid_versioning.parent.pid_value
+        else:
+            version_parent_pid_value = pid_value
+        records = []
+        pids_and_meta = \
+            db.session.query(PersistentIdentifier, RecordMetadata).join(
+                PIDRelation,
+                PIDRelation.child_id == PersistentIdentifier.id).\
+            filter(PIDRelation.parent_id ==
+                   RecordUUIDProvider.get(version_parent_pid_value).pid.id,
```

Comment:
This should be optimized by making an additional join to the PIDStore table. The current code is doing two requests to the DB instead of one: `RecordUUIDProvider.get(version_parent_pid_value)` + `JOIN query`